common/src/IO/FileSystem.h

```diff
@@ -38,14 +38,21 @@ namespace TrenchBroom {
         class FileSystem {
             deleteCopyAndMove(FileSystem)
         protected:
-            std::unique_ptr<FileSystem> m_next;
+            /**
+             * Next filesystem in the search path.
+             *
+             * NOTE: the use of std::shared_ptr is because there is shared ownership during construction of the
+             * filesystems (e.g. Quake3ShaderFileSystem needs to access m_next in its constructor, and the code
```

Comment:
I think it's more general: Any file system constructor might throw, and this would end up breaking the chain because the chain's tip was already moved into the new tip at the point of throwing. Shared pointer avoids this because the original pointer remains valid.